framework/include/interfacekernels/InterfaceKernel.h

```diff
@@ -28,33 +39,139 @@ InputParameters validParams<InterfaceKernel>();
  * InterfaceKernel is responsible for interfacing physics across subdomains
  */
 
-class InterfaceKernel : public DGKernelBase, public TwoMaterialPropertyInterface
+class InterfaceKernel : public MooseObject,
+                        public BoundaryRestrictable,
+                        public SetupInterface,
+                        public TransientInterface,
+                        public FunctionInterface,
+                        public UserObjectInterface,
+                        public NeighborCoupleableMooseVariableDependencyIntermediateInterface,
+                        public Restartable,
+                        public ZeroInterface,
+                        public MeshChangedInterface,
+                        public TwoMaterialPropertyInterface
 {
 public:
   InterfaceKernel(const InputParameters & parameters);
 
+  /*
+   * The master variable that this interface kernel operates on
+   */
+  MooseVariable & variable() const;
+
   /**
-   * The neighbor variable number that this kernel operates on
+   * The neighbor variable number that this interface kernel operates on
    */
   const MooseVariable & neighborVariable() const;
 
+  /**
+   * Return a reference to the subproblem.
+   */
+  SubProblem & subProblem();
+
   virtual void computeElemNeighResidual(Moose::DGResidualType type);
   virtual void computeElemNeighJacobian(Moose::DGJacobianType type);
-  virtual void computeJacobian();
   virtual void computeOffDiagElemNeighJacobian(Moose::DGJacobianType type, unsigned int jvar);
+
   virtual void computeElementOffDiagJacobian(unsigned int jvar);
   virtual void computeNeighborOffDiagJacobian(unsigned int jvar);
 
+  /**
+   * Computes the residual for the current side.
+   */
+  virtual void computeResidual();
+
+  /**
+   * Computes the jacobian for the current side.
+   */
+  virtual void computeJacobian();
+
 protected:
   virtual Real computeQpResidual(Moose::DGResidualType type) = 0;
   virtual Real computeQpJacobian(Moose::DGJacobianType type) = 0;
   virtual Real computeQpOffDiagJacobian(Moose::DGJacobianType type, unsigned int jvar);
+  /// The volume of the current neighbor
+  const Real & getNeighborElemVolume();
+
+  SubProblem & _subproblem;
```

Comment:
Since you are in here, can you document all this stuff with `///` and switch the other member docs that use `/**` to use `///`.